UnityProject/Assets/BulletKinetic.cs

```diff
@@ -0,0 +1,121 @@
+﻿using UnityEngine;
+using System.Collections;
+using System.Collections.Generic;
+public class BulletKinetic : BulletBehaviour
+{
+	public float maxBulletDistance;
+	//to control Kinetic animation with it
+	private bool isOnDespawn = false;
+
+	public override void Shoot(Vector2 dir, GameObject controlledByPlayer, Gun fromWeapon, BodyPartType targetZone = BodyPartType.Chest)
+	{
+		StartShoot(dir, controlledByPlayer, fromWeapon, targetZone);
```

Comment:
use `base.Shoot()` so all the default behaviour still happens
```suggestion
		base.Shoot(dir, controlledByPlayer, fromWeapon, targetZone);
```